integration_test/tests/bind_cols_tests.rs

```diff
@@ -0,0 +1,160 @@
+    /// After allocating a statement handle, the flow is:
+    ///     - SQLSetStmtAttrW(SQL_ATTR_ROW_ARRAY_SIZE, xxx)
+    ///     - SQLExecDirectW(<query>)
+    ///     - SQLBindCol
+    ///     - <loop: until SQLFetchScroll returns SQL_NO_DATA>
+    ///         - SQLFetchScroll
+    #[test]
+    fn test_bind_cols_and_fetch_next_rowset() {
+        let (env_handle, conn_handle, stmt_handle) =
+            default_setup_connect_and_alloc_stmt(AttrOdbcVersion::SQL_OV_ODBC3);
+
+        unsafe {
+            let row_array_size = 2;
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLSetStmtAttrW(
+                    stmt_handle,
+                    StatementAttribute::SQL_ATTR_ROW_ARRAY_SIZE as Integer,
+                    row_array_size as Pointer,
+                    0
+                ),
+                "{}",
+                get_sql_diagnostics(HandleType::SQL_HANDLE_STMT, stmt_handle as Handle)
+            );
+
+            exec_direct_default_query(stmt_handle);
+
+            // Since ROW_ARRAY_SIZE is set to 2, we double the buffer lengths.
+            let id_buffer = &mut [0u8; 8] as *mut _;
+            let id_indicator = &mut [0isize; 2] as *mut Len;
+            let a_buffer = &mut [0u8; 16] as *mut _;
+            let a_indicator = &mut [0isize; 2] as *mut Len;
+            bind_cols(
+                stmt_handle,
+                vec![
+                    (
+                        CDataType::SQL_C_SLONG,
+                        id_buffer as Pointer,
+                        4,
+                        id_indicator,
+                    ),
+                    (
+                        CDataType::SQL_C_SBIGINT,
+                        a_buffer as Pointer,
+                        8,
+                        a_indicator,
+                    ),
+                ],
+            );
+
+            // Data:
+            // - {_id: 0, a: {$numberLong: "42"}}
+            // - {_id: 1, a: {$numberLong: "13"}}
+            // - {_id: 2, a: {$numberLong: "100"}}
+            let mut i = 0;
+            let expected_id_data = [(0i32, 1i32), (2, 1)];
+            let expected_a_data = [(42i64, 13i64), (100, 13)];
+            loop {
+                let result =
+                    SQLFetchScroll(stmt_handle, FetchOrientation::SQL_FETCH_NEXT as SmallInt, 0);
+                if result == SqlReturn::NO_DATA {
+                    break;
+                }
+                assert_eq!(SqlReturn::SUCCESS, result);
+
+                // After fetching the data, check that the buffers contain the
+                // expected values. As shown above, the data only has 3 rows.
+                assert!(i < 2, "too much data fetched");
+
+                let (id1, id2) = expected_id_data[i];
+                assert_eq!(id1, *(id_buffer as *mut i32));
+                assert_eq!(id2, *((id_buffer as ULen + 4) as *mut i32));
+                assert_eq!(4, *id_indicator);
```

Comment:
You can reuse the transfer octet size here and for the a_buffer offset too if you decide to use it.
Also, this is an array too, so you should use offsets.

Reply:
> Also, this is an array too, so you should use offsets.

To clarify this, do you mean that since `id_indicator` is an array of length 2, I should ensure that each element, `id_indicator[0]` and `id_indicator[1]` is `ID_TRANSFER_OCTET_LENGTH`? (And do the same for `a`)

Reply:
I think what she means is that we should check both values in the indicator array instead of just the first one. Also, the indicator, will always be `isize`, so the offset is 8.

Reply:
```
assert_eq!(4, *id_indicator);
assert_eq!(4, *((id_indicator as ULen + 8) as *mut Len));
```

Reply:
Yes, because it's the indicator for each I believe or at least this what I see here : https://learn.microsoft.com/en-us/sql/odbc/reference/develop-app/column-wise-binding?view=sql-server-ver16

Reply:
> ```
> assert_eq!(4, *id_indicator);
> assert_eq!(4, *((id_indicator as ULen + 8) as *mut Len));


Given the change Natacha suggested above, `id_indicator` and `a_indicator` are statically `&mut [isize; 2]` (they are cast to `*mut Len` when argued to `SQLBindCol`). So instead of doing this manual offset, can't I just use array indexing, like this:
```
assert_eq!(ID_TRANSFER_OCTET_LEN, id_indicator[0] as usize);
assert_eq!(ID_TRANSFER_OCTET_LEN, id_indicator[1] as usize);
```
and equivalent for `a_indicator`?

Reply:
yeah that looks good!